### PR TITLE
Feat/dataload voucher

### PR DESCRIPTION
**Pull request by lghiggino**
Optimization for query looking for vouchers passing appContract and input index

### CODE REVIEW

File: internal/convenience/repository/voucher_repository_test.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 
 	"github.com/calindra/nonodo/internal/commons"
 	"github.com/calindra/nonodo/internal/convenience/model"
```
Comment (fabiooshiro) [inline internal/convenience/repository/voucher_repository_test.go:10]: We are importing the same package twice here

File: internal/reader/adapter_v1.go [hunk 2]
```diff
@@ -40,9 +41,18 @@ func NewAdapterV1(
 	if err != nil {
 		panic(err)
 	}
+	voucherRepository := &cRepos.VoucherRepository{
```
Comment (fabiooshiro) [inline internal/reader/adapter_v1.go:44]: It's not your fault, however, the container is responsible for creating repository instances...

File: internal/reader/adapter_v1.go [hunk 3]
```diff
@@ -174,6 +184,29 @@ func (a AdapterV1) GetVouchers(
...
+	loaders := loaders.For(ctx)
+	if loaders == nil {
+		return a.GetVouchers(ctx, nil, nil, nil, nil, inputIndex, nil)
+	} else {
+		appContract, err := getAppContractFromContext(ctx)
+		if err != nil {
+			return nil, err
+		}
+		key := repository.GenerateBatchVoucherKey(appContract, *inputIndex)
+		slog.Debug("VoucherLoader", "key", key)
```
Comment (fabiooshiro) [inline internal/reader/adapter_v1.go:197]: annoying log

File: internal/convenience/repository/voucher_repository_test.go (hunk 1 above)
Comment (lghiggino) [reply internal/convenience/repository/voucher_repository_test.go]: fixed and found another instance of same mistake

**Review by fabiooshiro — APPROVED** [review]

**Pull request merged**